src/AzureOpenAIProxy.Management/Services/ModelService.cs

```diff
@@ -55,10 +92,60 @@ public async Task DeleteOwnerCatalogAsync(Guid catalogId)
         await db.SaveChangesAsync();
     }
 
+    public async Task<OwnerCatalog> GetOwnerCatalogAsync(Guid catalogId)
+    {
+        var result = await db.OwnerCatalogs.FindAsync(catalogId);
+
+        if (result is null)
+        {
+            return null!;
+        }
+
+        string? endpointKey = await PostgresDecryptValue(result.EndpointKeyEncrypted);
+        string? endpointUrl = await PostgresDecryptValue(result.EndpointUrlEncrypted);
+
+        result.EndpointKey = endpointKey!;
+        result.EndpointUrl = endpointUrl!;
+
+        return result;
+    }
+
     public async Task<IEnumerable<OwnerCatalog>> GetOwnerCatalogsAsync()
     {
         string entraId = await authService.GetCurrentUserEntraIdAsync();
-        return await db.OwnerCatalogs.Where(oc => oc.Owner.OwnerId == entraId).OrderBy(oc => oc.FriendlyName).ToListAsync();
+        var catalogItems = await db.OwnerCatalogs.Where(oc => oc.Owner.OwnerId == entraId).OrderBy(oc => oc.FriendlyName).ToListAsync();
+
+        foreach (var catalog in catalogItems)
+        {
+            string? endpointUrl = await PostgresDecryptValue(catalog.EndpointUrlEncrypted);
+            catalog.EndpointUrl = endpointUrl!;
+        }
```

Comment:
This is pretty database intensive, since you have to hit the DB for every single item returned, resulting in a N^2 operation.

Either make it a PG Function and do the decryption on the DB server, or drop the endpoint from needing to be displayed.

Reply:
Hey I think it's only to the N ops. But have removed as the friendly name sort of serves the same purpose to help differentiate endpoints urls...
What I'd like to do, is for each resourse list which event is using. But not for today...